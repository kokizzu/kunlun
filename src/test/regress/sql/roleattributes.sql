-- default for superuser is false
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_superuser;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_superuser';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_superuser WITH SUPERUSER;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_superuser';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_superuser WITH NOSUPERUSER;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_superuser';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_superuser WITH SUPERUSER;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_superuser';

-- default for inherit is true
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_inherit;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_inherit';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_inherit WITH NOINHERIT;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_inherit';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_inherit WITH INHERIT;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_inherit';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_inherit WITH NOINHERIT;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_inherit';

-- default for create role is false
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_createrole;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_createrole';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_createrole WITH CREATEROLE;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createrole';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_createrole WITH NOCREATEROLE;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createrole';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_createrole WITH CREATEROLE;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createrole';

-- default for create database is false
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_createdb;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_createdb';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_createdb WITH CREATEDB;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createdb';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_createdb WITH NOCREATEDB;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createdb';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_createdb WITH CREATEDB;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createdb';

-- default for can login is false for role
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_role_canlogin;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_role_canlogin';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_role_canlogin WITH LOGIN;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_role_canlogin';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_role_canlogin WITH NOLOGIN;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_role_canlogin';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_role_canlogin WITH LOGIN;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_role_canlogin';

-- default for can login is true for user
--DDL_STATEMENT_BEGIN--
CREATE USER regress_test_def_user_canlogin;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_user_canlogin';
--DDL_STATEMENT_BEGIN--
CREATE USER regress_test_user_canlogin WITH NOLOGIN;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_user_canlogin';
--DDL_STATEMENT_BEGIN--
ALTER USER regress_test_user_canlogin WITH LOGIN;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_user_canlogin';
--DDL_STATEMENT_BEGIN--
ALTER USER regress_test_user_canlogin WITH NOLOGIN;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_user_canlogin';

-- default for replication is false
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_replication;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_replication';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_replication WITH REPLICATION;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_replication';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_replication WITH NOREPLICATION;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_replication';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_replication WITH REPLICATION;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_replication';

-- default for bypassrls is false
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_def_bypassrls;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_bypassrls';
--DDL_STATEMENT_BEGIN--
CREATE ROLE regress_test_bypassrls WITH BYPASSRLS;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_bypassrls';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_bypassrls WITH NOBYPASSRLS;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_bypassrls';
--DDL_STATEMENT_BEGIN--
ALTER ROLE regress_test_bypassrls WITH BYPASSRLS;
--DDL_STATEMENT_END--
SELECT * FROM pg_authid WHERE rolname = 'regress_test_bypassrls';

-- clean up roles
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_superuser;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_superuser;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_inherit;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_inherit;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_createrole;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_createrole;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_createdb;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_createdb;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_role_canlogin;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_role_canlogin;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP USER regress_test_def_user_canlogin;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP USER regress_test_user_canlogin;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_replication;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_replication;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_def_bypassrls;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE regress_test_bypassrls;
--DDL_STATEMENT_END--