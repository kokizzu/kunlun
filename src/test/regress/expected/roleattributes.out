-- default for superuser is false
CREATE ROLE regress_test_def_superuser;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_superuser';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_superuser | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_superuser WITH SUPERUSER;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_superuser';
        rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_superuser | t        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_superuser WITH NOSUPERUSER;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_superuser';
        rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_superuser | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_superuser WITH SUPERUSER;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_superuser';
        rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_superuser | t        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

-- default for inherit is true
CREATE ROLE regress_test_def_inherit;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_inherit';
         rolname          | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
--------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_inherit | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_inherit WITH NOINHERIT;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_inherit';
       rolname        | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_inherit | f        | f          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_inherit WITH INHERIT;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_inherit';
       rolname        | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_inherit | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_inherit WITH NOINHERIT;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_inherit';
       rolname        | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_inherit | f        | f          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

-- default for create role is false
CREATE ROLE regress_test_def_createrole;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_createrole';
           rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_createrole | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_createrole WITH CREATEROLE;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createrole';
         rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_createrole | f        | t          | t             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_createrole WITH NOCREATEROLE;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createrole';
         rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_createrole | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_createrole WITH CREATEROLE;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createrole';
         rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_createrole | f        | t          | t             | f           | f           | f              | f            |           -1 |             | 
(1 row)

-- default for create database is false
CREATE ROLE regress_test_def_createdb;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_createdb';
          rolname          | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
---------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_createdb | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_createdb WITH CREATEDB;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createdb';
        rolname        | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-----------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_createdb | f        | t          | f             | t           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_createdb WITH NOCREATEDB;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createdb';
        rolname        | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-----------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_createdb | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_createdb WITH CREATEDB;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_createdb';
        rolname        | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-----------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_createdb | f        | t          | f             | t           | f           | f              | f            |           -1 |             | 
(1 row)

-- default for can login is false for role
CREATE ROLE regress_test_def_role_canlogin;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_role_canlogin';
            rolname             | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
--------------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_role_canlogin | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_role_canlogin WITH LOGIN;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_role_canlogin';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_role_canlogin | f        | t          | f             | f           | t           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_role_canlogin WITH NOLOGIN;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_role_canlogin';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_role_canlogin | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_role_canlogin WITH LOGIN;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_role_canlogin';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_role_canlogin | f        | t          | f             | f           | t           | f              | f            |           -1 |             | 
(1 row)

-- default for can login is true for user
CREATE USER regress_test_def_user_canlogin;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_user_canlogin';
            rolname             | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
--------------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_user_canlogin | f        | t          | f             | f           | t           | f              | f            |           -1 |             | 
(1 row)

CREATE USER regress_test_user_canlogin WITH NOLOGIN;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_user_canlogin';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_user_canlogin | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER USER regress_test_user_canlogin WITH LOGIN;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_user_canlogin';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_user_canlogin | f        | t          | f             | f           | t           | f              | f            |           -1 |             | 
(1 row)

ALTER USER regress_test_user_canlogin WITH NOLOGIN;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_user_canlogin';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_user_canlogin | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

-- default for replication is false
CREATE ROLE regress_test_def_replication;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_replication';
           rolname            | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_replication | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_replication WITH REPLICATION;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_replication';
         rolname          | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
--------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_replication | f        | t          | f             | f           | f           | t              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_replication WITH NOREPLICATION;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_replication';
         rolname          | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
--------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_replication | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_replication WITH REPLICATION;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_replication';
         rolname          | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
--------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_replication | f        | t          | f             | f           | f           | t              | f            |           -1 |             | 
(1 row)

-- default for bypassrls is false
CREATE ROLE regress_test_def_bypassrls;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_def_bypassrls';
          rolname           | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
----------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_def_bypassrls | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

CREATE ROLE regress_test_bypassrls WITH BYPASSRLS;
CREATE ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_bypassrls';
        rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_bypassrls | f        | t          | f             | f           | f           | f              | t            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_bypassrls WITH NOBYPASSRLS;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_bypassrls';
        rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_bypassrls | f        | t          | f             | f           | f           | f              | f            |           -1 |             | 
(1 row)

ALTER ROLE regress_test_bypassrls WITH BYPASSRLS;
ALTER ROLE
SELECT * FROM pg_authid WHERE rolname = 'regress_test_bypassrls';
        rolname         | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
------------------------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
 regress_test_bypassrls | f        | t          | f             | f           | f           | f              | t            |           -1 |             | 
(1 row)

-- clean up roles
DROP ROLE regress_test_def_superuser;
DROP ROLE
DROP ROLE regress_test_superuser;
DROP ROLE
DROP ROLE regress_test_def_inherit;
DROP ROLE
DROP ROLE regress_test_inherit;
DROP ROLE
DROP ROLE regress_test_def_createrole;
DROP ROLE
DROP ROLE regress_test_createrole;
DROP ROLE
DROP ROLE regress_test_def_createdb;
DROP ROLE
DROP ROLE regress_test_createdb;
DROP ROLE
DROP ROLE regress_test_def_role_canlogin;
DROP ROLE
DROP ROLE regress_test_role_canlogin;
DROP ROLE
DROP USER regress_test_def_user_canlogin;
DROP ROLE
DROP USER regress_test_user_canlogin;
DROP ROLE
DROP ROLE regress_test_def_replication;
DROP ROLE
DROP ROLE regress_test_replication;
DROP ROLE
DROP ROLE regress_test_def_bypassrls;
DROP ROLE
DROP ROLE regress_test_bypassrls;
DROP ROLE
